rules/no-console-spaces.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+const getDocsUrl = require('./utils/get-docs-url');
+
+const getConsoleMethod = node => {
+	const methods = [
+		'log',
+		'debug',
+		'info',
+		'warn',
+		'error'
+	];
+
+	const {callee} = node;
+
+	if (
+		callee.type === 'MemberExpression' &&
+		callee.object.type === 'Identifier' &&
+		callee.object.name === 'console' &&
+		callee.property.type === 'Identifier' &&
+		methods.includes(callee.property.name)
+	) {
+		return callee.property.name;
+	}
+};
+
+const getArgumentValue = (context, nodeArgument) => {
+	let value = null;
+
+	if (nodeArgument.type === 'Literal' && typeof nodeArgument.value === 'string') {
+		value = nodeArgument.value;
+	}
+
+	if (nodeArgument.type === 'TemplateLiteral') {
+		const sourceCode = context.getSourceCode();
+		value = sourceCode.getText(nodeArgument);
+		// Strip off backticks
+		value = value.substring(1, value.length - 1);
+	}
+
+	return value;
+};
+
+const fixValue = value => {
+	if (!value) {
+		return value;
+	}
+
+	// Find exactly one leading or tailing space
+	return value.replace(/^ ?((?! ).*?[^ ]) ?$/, '$1');
```

Comment:
Seems like using `.startsWith()`/`.endsWith()` and then `.slice(1)` or `.slice(-1)` would be simpler and more readable.